Dependencies/xr/Source/ARCore/XR.cpp

```diff
@@ -860,6 +862,69 @@ namespace xr
             }
         }
 
+        void UpdateFeaturePointCloud()
+        {
+            if (!IsTracking() || !FeaturePointCloudEnabled)
+            {
+                return;
+            }
+
+            // Get the feature point cloud from ArCore.
+            ArPointCloud *pointCloud = nullptr;
+            int32_t numberOfPoints = 0;
+            const int32_t* pointCloudIDs = nullptr;
+            const float *pointCloudData = nullptr;
+            ArStatus status = ArFrame_acquirePointCloud(session, frame, &pointCloud);
+
+            if (status != AR_SUCCESS)
+            {
+                FeaturePointCloud.clear();
+                return;
+            }
+
+            try
+            {
+                ArPointCloud_getNumberOfPoints(session, pointCloud, &numberOfPoints);
+                ArPointCloud_getData(session, pointCloud, &pointCloudData);
+                ArPointCloud_getPointIds(session, pointCloud, &pointCloudIDs);
+
+                FeaturePointCloud.resize(numberOfPoints);
+                for (int32_t i = 0; i < numberOfPoints; i++)
+                {
+                    FeaturePointCloud.emplace_back();
+                    auto& featurePoint = FeaturePointCloud.back();
+
+                    int32_t dataIndex = i * 4;
+                    featurePoint.x = pointCloudData[dataIndex];
+                    featurePoint.y = pointCloudData[dataIndex + 1];
+                    featurePoint.z = -1 * pointCloudData[dataIndex + 2];
```

Comment:
Why is this value multiplied by -1?

Reply:
See the above comment from Hendry.  We transport Babylon space directly into camera space whereas this is reported directly in world space if you take a look at the documentation for ARCore getDisplayOrientedPose you will see that it is oriented such that -z is pointing in the direction the camera is looking. So we need to flip this position to be correct relative to its actual position.